File: app/src/main/java/net/mmho/photomap2/QueryBuilder.java
package net.mmho.photomap2;

import com.google.android.gms.maps.model.LatLng;
import com.google.android.gms.maps.model.LatLngBounds;

import static android.provider.MediaStore.Images.ImageColumns.LATITUDE;
import static android.provider.MediaStore.Images.ImageColumns.LONGITUDE;
import static android.provider.MediaStore.Images.ImageColumns.DATE_TAKEN;
import static android.provider.MediaStore.Images.ImageColumns._ID;

class QueryBuilder {
    static String createQuery(LatLngBounds bounds){
        LatLng start = bounds.southwest;
        LatLng end = bounds.northeast;
        String latitude = String.format("%s between %s and %s and ",
                LATITUDE,Double.toString(start.latitude),Double.toString(end.latitude));
        String longitude;
        if(start.longitude<end.longitude){
            longitude= String.format("%s between %s and %s",
                    LONGITUDE,Double.toString(start.longitude),Double.toString(end.longitude));
        }
        else{
            longitude = String.format("(%s between -180.0 and %s or %s between %s and 180.0)",
                    LONGITUDE,Double.toString(end.longitude),LONGITUDE,Double.toString(start.longitude));

        }

        return latitude+longitude;

    }

    static String createQuery(long id){
        return _ID+" is "+id;
    }

    static String createQuery(){
        return LATITUDE+" not null and "+LONGITUDE+" not null";
    }

    static String createQueryNoLocation(){
        return LATITUDE+" is null or "+LONGITUDE+"is null";
    }

    static String sortDateNewest(){
        return DATE_TAKEN+" desc";
    }

    static String sortDateOldest() {
        return DATE_TAKEN+" asc";
    }

}

File: app/src/main/kotlin/net/mmho/photomap2/QueryBuilder.kt
package net.mmho.photomap2

import android.provider.BaseColumns
import android.provider.MediaStore.Images.ImageColumns.*
import com.google.android.gms.maps.model.LatLngBounds
import java.util.*

internal object QueryBuilder {
    fun createQuery(bounds: LatLngBounds): String {
        val start = bounds.southwest
        val end = bounds.northeast
        val latitude = "$LATITUDE between %f and %f and ".format(Locale.US,start.latitude,end.latitude)
        val longitude: String =
        if (start.longitude < end.longitude)
            "$LONGITUDE between %f and %f".format(Locale.US,start.longitude,end.longitude)
        else "($LONGITUDE between -180.0 and %f or $LONGITUDE between %f and 180.0)"
            .format(Locale.US,end.longitude, LONGITUDE,start.longitude)

        return latitude + longitude

    }

    fun createQuery(id: Long): String {
        return "${BaseColumns._ID} is $id"
    }

    fun createQuery(): String {
        return "$LATITUDE not null and $LONGITUDE not null"
    }

    fun createQueryNoLocation(): String {
        return "$LATITUDE is null or $LONGITUDE is null"
    }

    fun sortDateNewest(): String {
        return "$DATE_TAKEN desc"
    }

    fun sortDateOldest(): String {
        return "$DATE_TAKEN asc"
    }
}
